components/logins/src/error.rs

```diff
@@ -97,193 +121,70 @@ pub enum InvalidLogin {
+            Self::NonEmptyTable => ErrorHandling::unexpected(
+                LoginsStorageError::UnexpectedLoginsStorageError(
+                    "must be an empty DB to migrate".to_string(),
+                ),
+                Some("migration"),
+            ),
+            Self::CryptoError(_) => {
+                ErrorHandling::log(LoginsStorageError::IncorrectKey, log::Level::Warn)
             }
-            // We can't destructure `err` without bringing in the libsqlite3_sys crate
-            // (and I'd really rather not) so we can't put this in the match.
-            ErrorKind::SqlError(rusqlite::Error::SqliteFailure(err, _))
-                if err.code == rusqlite::ErrorCode::NotADatabase =>
-            {
-                log::error!("Not a database / invalid key error");
-                LoginsStorageError::InvalidKey(label)
-            }
-
-            ErrorKind::InvalidKey => LoginsStorageError::InvalidKey(label),
-
-            ErrorKind::SqlError(rusqlite::Error::SqliteFailure(err, _))
-                if err.code == rusqlite::ErrorCode::OperationInterrupted =>
-            {
-                log::warn!("Operation interrupted (SQL)");
-                LoginsStorageError::Interrupted(label)
-            }
-
-            ErrorKind::Interrupted(_) => {
-                log::warn!("Operation interrupted (Outside SQL)");
-                LoginsStorageError::Interrupted(label)
-            }
-
-            ErrorKind::InvalidSalt => {
-                log::error!("Invalid salt provided");
-                // In the old world, this had an error code (7) but no Kotlin
-                // error type, meaning it got the "base" error.
-                LoginsStorageError::UnexpectedLoginsStorageError(label)
-            }
-
-            ErrorKind::CryptoError(_) => {
-                log::error!("Crypto error");
-                LoginsStorageError::CryptoError(label)
-            }
-
-            err => {
-                log::error!("UnexpectedLoginsStorageError error: {:?}", err);
-                LoginsStorageError::UnexpectedLoginsStorageError(label)
+            Self::Interrupted(_) => {
+                ErrorHandling::passthrough(LoginsStorageError::Interrupted(self.to_string()))
             }
+            Self::SyncAdapterError(e) => match e.kind() {
+                Sync15ErrorKind::TokenserverHttpError(401) | Sync15ErrorKind::BadKeyLength(..) => {
+                    ErrorHandling::log(
+                        LoginsStorageError::SyncAuthInvalid(e.to_string()),
+                        log::Level::Warn,
+                    )
+                }
+                Sync15ErrorKind::RequestError(_) => ErrorHandling::log(
+                    LoginsStorageError::RequestFailed(e.to_string()),
+                    log::Level::Warn,
+                ),
+                _ => ErrorHandling::unexpected(
+                    LoginsStorageError::UnexpectedLoginsStorageError(self.to_string()),
+                    Some("sync"),
```

Comment:
Should this and other error classification strings be prefixed with `logins-`?  That way when we add more components, each gets its own sync error classification?